### PR TITLE
Reaper now reports if the target resource has been deleted. 

### PR DESCRIPTION
**Pull request by iocanel**
Long story short: When deleting a `dc` the rc is not removed. It seems that Openshift will not remove the `rc` if `dc` is still there *(This also applies to oc)*.

So we change the order of things and we delete the DC from within the reaper.

### CODE REVIEW

**Comment by iocanel** [comment]
@jimmidyson: Can you have a look please?

File: kubernetes-examples/src/main/java/io/fabric8/openshift/examples/DeploymentConfigExamples.java [hunk 1]
```diff
@@ -76,11 +88,13 @@ public static void main(String[] args) throws InterruptedException {
...
       log("Created pods:", client.pods().inNamespace("thisisatest").list().getItems());
-      client.deploymentConfigs().inNamespace("thisisatest").withName("nginx").scale(0);
+      client.deploymentConfigs().inNamespace("thisisatest").withName("nginx").delete();
+      log("Pods:", client.pods().inNamespace("thisisatest").list().getItems());
+      log("Replication Controllers:", client.replicationControllers().inNamespace("thisisatest").list().getItems());
 
       log("Done.");
     }finally {
-      client.namespaces().withName("thisisatest").delete();
+     // client.projects().withName("thisisatest").delete();
```
Comment (jimmidyson) [inline kubernetes-examples/src/main/java/io/fabric8/openshift/examples/DeploymentConfigExamples.java]: Why not delete this?

**Comment by jimmidyson** [comment]
Let's merge this for now & look at async GC later.

**Comment by jimmidyson** [comment]
[merge]

**Pull request merged**